src/main/java/org/example/controllers/JobRoleController.java

```diff
@@ -31,4 +32,16 @@ public Response getListOpenJobRoles() {
             return Response.serverError().build();
         }
     }
+
+    @GET
+    @Path("/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getJobRoleById(@PathParam("id") final int id) {
+        try {
+            return Response.ok()
+                    .entity(jobRoleService.getJobRoleById(id)).build();
+        } catch (JobRoleDaoException e) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
```

Comment:
have a little think about this. What if the database is down for whatever reason? 404 probably doesn't make sense? Personally I would:

1. put this back to how it was
2. move jobRoleService.getJobRoleById(id) to a variable inside the try
3. Add an if to check the jobRoleService object - if it's missing an id, it's a 404, if it's valid it's a 200
4. We also need to think about if an id is not an id it's a 400 bad request. Maybe because we're mapping it to an int it already works, but lets make sure we test this in integration tests. In there we should be able to say hey, if we call this url, what is the response?

then there's a lot more tests to write

Reply:
Fixed I think! Let me know if the way I implemented this is okay